pkg/controllers/user/systemimage/systemimage.go

```diff
@@ -0,0 +1,142 @@
+package systemimage
+
+import (
+	"crypto/md5"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/rancher/norman/types/convert"
+	"github.com/rancher/rancher/pkg/controllers/user/workload"
+	"github.com/rancher/rancher/pkg/settings"
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+const (
+	systemImageVersionAnn = "field.cattle.io/systemImageVersion"
+	systemProjectAnn      = "management.cattle.io/system-namespace"
+)
+
+var systemProjectLabels = labels.Set(map[string]string{"authz.management.cattle.io/system-project": "true"})
+
+type Syncer struct {
+	clusterName        string
+	projectLister      v3.ProjectLister
+	projects           v3.ProjectInterface
+	workloadController workload.CommonController
+}
+
+func (s *Syncer) Sync(key string, obj *v3.Project) error {
+	projects, err := s.projectLister.List(s.clusterName, systemProjectLabels.AsSelector())
+	if err != nil {
+		return fmt.Errorf("get projects failed, %v", err)
+	}
+
+	var systemProject *v3.Project
+	for _, v := range projects {
+		if v.Spec.DisplayName == "System" {
+			systemProject = v.DeepCopy()
+		}
+	}
+
+	if systemProject == nil {
+		return nil
+	}
+
+	latestSysImageVersion, err := GetSystemImageVersion()
+	if err != nil {
+		return fmt.Errorf("get latest systemImageVersion failed: %v", err)
+	}
+
+	curSysImageVersion := systemProject.Annotations[systemImageVersionAnn]
+	if curSysImageVersion == latestSysImageVersion {
+		return nil
+	}
+
+	systemImageMap, err := getSystemImagesMap(v3.ToolsSystemImages)
+	if err != nil {
+		return fmt.Errorf("getSystemImagesMap failed, %v", err)
+	}
+
+	nss := settings.SystemNamespaces.Default
+	namespaces := strings.Split(nss, ",")
+	for _, n := range namespaces {
+		wls, err := s.workloadController.GetAllWorkloads(n)
```

Comment:
This lists all workloads. We may get a deployment and its replica set simultaneously. Should we list specific workload types or list the ones that have no owner reference.